src/modeler/AutoRest.Swagger/Validation/OperationDescriptionRequired.cs

```diff
@@ -1,34 +1,24 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-using AutoRest.Core.Logging;
-using AutoRest.Core.Properties;
-using AutoRest.Swagger.Validation.Core;
 using AutoRest.Swagger.Model;
+using AutoRest.Swagger.Validation.Core;
 
 namespace AutoRest.Swagger.Validation
 {
-    public class OperationDescriptionRequired : TypedRule<Operation>
+    public class OperationDescriptionRequired : DescriptionRequired<Operation>
     {
         /// <summary>
         /// This rule fails if the description is null and the reference is null (since the reference could have a description)
         /// </summary>
-        /// <param name="entity"></param>
-        /// <returns></returns>
-        public override bool IsValid(Operation entity)
-            => entity == null || entity.Description != null;
-
-        /// <summary>
-        /// The template message for this Rule. 
-        /// </summary>
-        /// <remarks>
-        /// This may contain placeholders '{0}' for parameterized messages.
-        /// </remarks>
-        public override string MessageTemplate => Resources.MissingDescription;
-
-        /// <summary>
-        /// The severity of this message (ie, debug/info/warning/error/fatal, etc)
-        /// </summary>
-        public override Category Severity => Category.Warning;
+        /// <param name="entity">Entity being validated</param>
+        /// <param name="context">Rule context</param>
+        /// <param name="formatParameters">formatted parameters</param>
+        /// <returns><c>true</c> if entity contains description, <c>false</c> otherwise</returns>
+        public override bool IsValid(Operation entity, RuleContext context, out object[] formatParameters)
+        {
+            formatParameters = new string[] { string.Format("operation '{0}'", entity.OperationId) };
+            return entity?.Description != null;
```

Comment:
should this condition be the same as the one for "parameters" "return entity?.Description != null || !string.IsNullOrWhiteSpace(entity.Reference);"